cache/ttl_cache.ts

```diff
@@ -0,0 +1,97 @@
+// Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
+import type { MemoizationCache } from "./memoize.ts";
+import { delay } from "@std/async/delay";
+
+/**
+ * Time-to-live cache.
+ *
+ * Automatically removes entries once the configured amount of time elapses. If
+ * the values themselves are promises, the countdown starts from when they
+ * resolve.
+ *
+ * @typeParam K The type of the cache keys.
+ * @typeParam V The type of the cache values.
+ *
+ * @example Usage
+ * ```ts
+ * import { TtlCache } from "@std/cache";
+ * import { assertEquals } from "@std/assert";
+ * import { delay } from "@std/async/delay";
+ *
+ * const cache = new TtlCache<string, number>(1000);
+ *
+ * cache.set("a", 1);
+ * assertEquals(cache.size, 1);
+ * await delay(2000);
+ * assertEquals(cache.size, 0);
+ * ```
+ */
+export class TtlCache<K, V> extends Map<K, V>
+  implements MemoizationCache<K, V> {
+  /**
+   * The default time-to-live in milliseconds
+   *
+   * @example Usage
+   * ```ts
+   * import { TtlCache } from "@std/cache";
+   * import { assertEquals } from "@std/assert";
+   * const cache = new TtlCache(1000);
+   * assertEquals(cache.defaultTtl, 1000);
+   * ```
+   */
+  defaultTtl: number;
```

Comment:
I think it's best that we don't expose any properties unless we have a real good reason to. I think it's fine if this is private.
```suggestion
  #defaultTtl: number;
```